### ISSUE TITLE
Fails to detect header in div

### ISSUE THREAD

**Issue by agx** [markdownlint/markdownlint#534]
## Description

This example

```
<div class="book-hero">

# Foo Contributors Manual

</div>

## Whatever
```

Let's mdl throw

> MD002 First header should be a top level header

although there is a top level header. Removing the `div` makes it go away.

### Environment

0.13.0

### Expected Behavior

The toplevel header should be detected

### Actual Behavior

Toplevel header isn't detected

## Replication Case

See above. 

(Thanks for this useful tool btw)

**Comment by nbehrnd** [comment]
@agx As a bypass, you might consider _to mute_ rule MD002 as shown in Debian's [man page]:

```bash
mdl -r ~MD002 example.md
```

[man page]: https://manpages.debian.org/trixie/markdownlint/markdownlint.1.en.html

**Comment by agx** [comment]
Thanks for the hint. I've had done that already (via config file).
